utils/sendGridEventWebhook.js

```diff
@@ -1,39 +1,56 @@
 const sgMail = require("@sendgrid/mail");
 const { SecretManagerServiceClient } = require("@google-cloud/secret-manager");
 const { EventWebhook, EventWebhookHeader } = require("@sendgrid/eventwebhook");
+const uuid = require('uuid')
+const admin = require('firebase-admin');
+const db = admin.firestore();
 const { processEventWebhook } = require("./firestore");
 const { getResponseJSON } = require("./shared");
 
+
+const _getSecrets = async (envName) => {
+    const client = new SecretManagerServiceClient();
+    const [version] = await client.accessSecretVersion({
+        name: envName
+    });
+    return version.payload.data.toString();
+};
+
 const _testSendEmail = async (res) => {
     console.log("Start sending email");
 
     try {
-        // Add SendGrid apiKey to test
-        sgMail.setApiKey("");
+        const notificationRecord = {
+            notificationSpecificationsID: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
+            id: uuid(),
+            notificationType: 'email',
+            email: '', // Update your data to test
+            notification: {
+              title: 'Test SendGrid Event Webhook',
+              body: '<p>Hello World!</p>',
+              time: new Date().toISOString(),
+            },
+            attempt: '3rd contact',
+            category: 'sendgrid',
+            token: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
+            uid: 't66mekkmYaPMg4hpSlGHqVKaIuB3',
+            uid: 'xxxxxxxxxxxxxxxxxxxxxxxxxxxx',
```

Comment:
looks like duplicate keys - should the first one be removed?

Reply:
Solved